### ISSUE TITLE
How to use textattack on generative models?

### ISSUE THREAD

**Issue by pratik2358** [QData/TextAttack#725]

**Comment by jxmorris12** [comment]
Hi, we have several examples of this in our documentation, plus the seq2sick recipe which allows you to run an adversarial attack against a single sequence-to-sequence model output. If you have a more specific question I can try to help you.

**Comment by StarWatch27** [comment]
> Hi, we have several examples of this in our documentation, plus the seq2sick recipe which allows you to run an adversarial attack against a single sequence-to-sequence model output. If you have a more specific question I can try to help you.

Hello, I've read the API doc and only find  one commandLine example using seq2sick. [link](https://textattack.readthedocs.io/en/master/3recipes/attack_recipes_cmd.html)
I need to attack BART in en-de translation task by using seq2sick and morpheus .
It would be great if you could provide the links for those examples in documentation, Thanks!

**Comment by qiyanjun** [comment]
Following up on @StarWatch27's more specific question above (attacking BART on en-de translation with seq2sick/morpheus) — that wasn't answered and is a genuinely different ask than the original issue. Leaving this open; if a maintainer or anyone in the community has direct experience running seq2sick against a seq2seq translation model, this is the open question.

**Comment by qiyanjun** [comment]
Following up on the open question about attacking a seq2seq translation model (e.g. BART) with `seq2sick`/`morpheus` — this now works and is documented directly in the recipe docstrings.

**Background:** the underlying blocker was that `HuggingFaceModelWrapper` only ran a plain forward pass, which returns logits, not text — so any text-to-text goal function (`NonOverlappingOutput` for seq2sick, `MinimizeBleu` for morpheus) had nothing to compare against for a raw `transformers` encoder-decoder model. That's fixed (#771): the wrapper now calls `.generate()` + decodes whenever the model is `is_encoder_decoder=True` and generation-capable, so it works uniformly across BART, MarianMT, mBART, T5, etc.

**For translation specifically**, use `MorpheusTan2020` with `MinimizeBleu` — it compares the model's output against a *reference* translation you supply, which is the natural setup for MT (as opposed to seq2sick's `NonOverlappingOutput`, which compares against the model's own unperturbed output — more suited to summarization where you don't have a ground-truth reference).

Worked example (verified end-to-end; using `t5-small` here since it's small and this exact snippet was tested — swap in a BART/MarianMT/mBART translation checkpoint the same way):

```python
import transformers
from textattack.attack_recipes import MorpheusTan2020
from textattack.models.wrappers import HuggingFaceModelWrapper

model = transformers.AutoModelForSeq2SeqLM.from_pretrained("t5-small")
tokenizer = transformers.AutoTokenizer.from_pretrained("t5-small")
# `max_length` avoids `transformers`' 20-token generation default, which
# truncates sentence-length translations.
model_wrapper = HuggingFaceModelWrapper(model, tokenizer, max_length=200)

attack = MorpheusTan2020.build(model_wrapper)
# t5-small needs the task prefix; other translation checkpoints
# (BART/MarianMT/mBART) typically don't.
input_text = "translate English to German: The quick brown fox jumps over the lazy dog."
reference_translation = "Der schnelle braune Fuchs springt über den faulen Hund."
result = attack.attack(input_text, reference_translation)
print(result.__str__(color_method="ansi"))
```

For a BART/MarianMT en-de checkpoint, load it the same way (`transformers.AutoModelForSeq2SeqLM.from_pretrained("Helsinki-NLP/opus-mt-en-de")` or an mBART checkpoint), drop the `"translate English to German: "` prefix (that's T5-specific), and pass a real reference translation for your input sentence.

If you'd rather compare against the model's own unperturbed translation instead of a fixed reference (e.g. you don't have gold references for your inputs), `Seq2SickCheng2018BlackBox` works the same way — see its docstring for that variant.

Docs: see the `MorpheusTan2020` and `Seq2SickCheng2018BlackBox` docstrings for the full examples (also rendered at https://textattack.readthedocs.io under Attack Recipes).

**Comment by qiyanjun** [comment]
Opened #840 with these docs changes.
